packages/task/src/node/task-server.slow-spec.ts

```diff
@@ -220,29 +224,68 @@ describe('Task server / back-end', function () {
         }
     });
 
-    it('task using raw process can be killed', async function () {
-        // const command = isWindows ? command_absolute_path_long_running_windows : command_absolute_path_long_running;
+    it('task using type "process" can be killed', async function () {
         const taskInfo: TaskInfo = await taskServer.run(createTaskConfigTaskLongRunning('process'), wsRoot);
 
-        const p = new Promise<string>((resolve, reject) => {
+        const p = new Promise<string | number>((resolve, reject) => {
             taskWatcher.onTaskExit((event: TaskExitedEvent) => {
-                if (event.taskId !== taskInfo.taskId || event.signal === undefined) {
-                    reject();
+                if (isWindows) {
+                    if (event.taskId !== taskInfo.taskId || event.code === undefined) {
+                        reject(new Error(JSON.stringify(event)));
+                    }
+                    resolve(event.code);
+                } else {
+                    if (event.taskId !== taskInfo.taskId || event.signal === undefined) {
+                        reject(new Error(JSON.stringify(event)));
+                    }
+                    resolve(event.signal);
                 }
-
-                resolve(event.signal);
             });
 
             taskServer.kill(taskInfo.taskId);
         });
 
-        const signal = await p;
-        expect(signal).equals('SIGTERM');
+        // node-pty reports different things on Linux/macOS vs Windows when
+        // killing a process.  This is not ideal, but that's how things are
+        // currently.  Ideally, its behavior should be aligned as much as
+        // possible on what node's child_process module does.
+        const signalOrCode = await p;
+        if (isWindows) {
+            // On Windows, node-pty just reports an exit code of 0.
+            expect(signalOrCode).equals(0);
+        } else {
+            // On Linux/macOS, node-pty sends SIGHUP by default, for some reason.
+            expect(signalOrCode).equals('SIGHUP');
+        }
     });
 
+    /**
+     * TODO: Figure out how to debug a process that correctly starts but exits with a return code > 0
+     */
     it('task using terminal process can handle command that does not exist', async function () {
```

Comment:
This test fails when I run the unit test  suite on Gitpod; `yarn test:theia` :

13 passing (3s)
  1 failing

  1) Task server / back-end task using terminal process can handle command that does not exist:
     Error: {"taskId":0,"ctx":"/workspace/theia/packages/task/lib/node/../../test-resources","signal":"SIGHUP"}
      at lib/node/task-server.slow-spec.js:399:44
      at /workspace/theia/packages/core/lib/common/event.js:161:33
      at CallbackList.invoke (/workspace/theia/packages/core/lib/common/event.js:176:39)
      at Emitter.fire (/workspace/theia/packages/core/lib/common/event.js:264:29)
      at Object.onTaskExit (lib/common/task-watcher.js:39:29)
      at lib/node/task-server.js:161:20
      at Array.forEach (<anonymous>)
      at TaskServerImpl.fireTaskExitedEvent (lib/node/task-server.js:160:22)
      at lib/node/task-server.js:140:35
      at /workspace/theia/packages/core/lib/common/event.js:161:33
      at CallbackList.invoke (/workspace/theia/packages/core/lib/common/event.js:176:39)
      at Emitter.fire (/workspace/theia/packages/core/lib/common/event.js:264:29)
      at ProcessTask.Task.fireTaskExited (lib/node/task.js:47:26)
      at lib/node/process/process-task.js:59:19
      at /workspace/theia/packages/core/lib/common/event.js:161:33
      at CallbackList.invoke (/workspace/theia/packages/core/lib/common/event.js:176:39)
      at Emitter.fire (/workspace/theia/packages/core/lib/common/event.js:264:29)
      at TerminalProcess.Process.emitOnExit (/workspace/theia/packages/process/lib/node/process.js:123:26)
      at PipeSocket.<anonymous> (/workspace/theia/packages/process/lib/node/terminal-process.js:106:27)
      at UnixTerminal.Terminal.emit (/workspace/theia/node_modules/@theia/node-pty/lib/terminal.js:79:34)
      at PipeSocket.<anonymous> (/workspace/theia/node_modules/@theia/node-pty/lib/unixTerminal.js:85:27)
      at Pipe._handle.close [as _onclose] (net.js:561:12)

Reply:
It was passing before I added the `shell || process.env["SHELL"] || 'sh'`, no idea why it complains now, will fix!

Reply:
Ok, so after more investigation, the failure seems to appear at random, that's most likely a race condition.

No real clue why yet.

Reply:
Ah, I forgot to remove a line that was killing the task (copy/paste from another test).

So it was indeed a race condition due to this line trying to kill the process, vs the task just failing.